usermods/pwm_outputs/readme.md

```diff
@@ -0,0 +1,21 @@
+# PWM outputs
+
+v2 Usermod to add generic PWM outputs to WLED. Usermode could be used to control servo motors, LED brightness or any other device controlled by PWM signal.
+
+## Installation
+
+Add the compile-time option `-D USERMOD_PWM_OUTPUTS` to your `platformio.ini` (or `platformio_override.ini`). By default upt to 3 PWM outputs could be configured, to increase that limit add build argument `-D USERMOD_PWM_OUTPUT_PINS=10` (replace 10 by desired amount).
+
+Currently only ESP32 is supported.
+
+## Configuration
+
+By default PWM outputs are disabled, navigate to Usermods settings and configure desired PWM pins and frequencies.
+
+## Usage
+
+If PWM output is configured, it starts to publish its duty cycle value (0-1) both to state JSON and to info JSON (visible in UI info panel). To set PWM duty cycle, use JSON api (over HTTP or over Serial)
+
+```json
+{"pwm_0": 0.0, "pwm_1": 0.1, ...}
```

Comment:
Perhaps converting this to an array of objects would be better. I.e.: `{"PwmOutputs":[{"id":0,"duty":0.1},{...},...]}`
This would allow some *ugly* constructs below.

Reply:
Thank you for you comments! Separated PWM states to separate object:
```json
{"pwm": {"0": {"duty": 0.5123}, "1": {"duty": 0.1}}}
``` 
Used `pwm` instead of `PwmOutputs` to match current style. Also, used `id` as keys, as it is much easier to get current duty cycle value for specific output (e.g `state['pwm']['0']['duty']`)

Reply:
I am not sure this will work with usermod settings implementation. Test it thoroughly, please.

Reply:
Yes, was not sure either, but tested it and everything works as expected. As a bonus, also pin conflict detection (red background at settings page) works.